### PR TITLE
feat: simple defi vault

### LINKED ISSUE #18: feat: Defi - simple Vault
Write a simple example of a Vault contract. You can simply translate this [Solidity implementation](https://solidity-by-example.org/defi/vault/) to Cairo

### PR DESCRIPTION
**Pull request by julio4**
Close #18

### CODE REVIEW

**Comment by enitrat** [comment]
We might need to give credit to https://solidity-by-example.org/ for this implementation

**Comment by enitrat** [comment]
> Yes for sure. Or do you want to make a slightly different version?

I like this one, it's easy to understand what's happening

File: src/ch01-02-simple_vault.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Simple Defi Vault
+
+This is the Cairo adaptation of the [Solidity by example Vault](https://solidity-by-example.org/defi/vault/).
+Here's how it works:
+
+- When a user deposits a token, the contract calculates the amount of shares to mint.
+
+- When an user withdraws, the contract burns their shares, calculates the yield, and withdraw both the yield and the initial amount of token deposited.
```
Comment (enitrat) [inline src/ch01-02-simple_vault.md:8]: > an user
a user

File: listings/ch01-applications/simple_vault/src/simple_vault.cairo [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+mod SimpleVault {
+    use super::{IERC20Dispatcher, IERC20DispatcherTrait};
+    use starknet::{
+        ContractAddress,
+        get_caller_address,
+        get_contract_address
+    };
+
+    struct Storage {
+        token: ContractAddress,
```
[the remark this answers is not in the document]
Comment (julio4) [reply listings/ch01-applications/simple_vault/src/simple_vault.cairo:43]: Great idea, let's do that!

**Review by enitrat — APPROVED** [review]
LGTM!

**Pull request merged**